docs/eloquent-models/model-class.txt

```diff
@@ -0,0 +1,292 @@
+   :language: php
+   :dedent:
+
+To learn more about customizing a Laravel authentication user provider,
+see `Adding Custom User Providers <https://laravel.com/docs/{+laravel-docs-version+}/authentication#adding-custom-user-providers>`__
+in the Laravel docs.
+
+.. _laravel-model-customize:
+
+Customize an Eloquent Model Class
+---------------------------------
+
+This section shows how to perform the following Eloquent model behavior
+customizations:
+
+- :ref:`laravel-model-customize-collection-name`
+- :ref:`laravel-model-customize-primary-key`
+- :ref:`laravel-model-soft-delete`
+- :ref:`laravel-model-cast-data-types`
+- :ref:`laravel-model-mass-assignment`
+
+.. _laravel-model-customize-collection-name:
+
+Change the Model Collection Name
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+To customize the name of the collection the model uses to retrieve and save
+data in MongoDB, override the ``$collection`` property of the model
+class.
+
+By default, the model uses the MongoDB database name set in your Laravel
+application's ``config/database.php`` setting and the snake case plural form of
+your model class name for the collection.
+
+The following example specifies the custom MongoDB collection name,
+``celestial_body``, for the ``Planet`` class:
+
+.. literalinclude:: /includes/eloquent-models/PlanetCollection.php
+   :language: php
+   :emphasize-lines: 9
+   :dedent:
+
+.. _laravel-model-customize-primary-key:
+
+Change the Primary Key Field
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+To customize the model's primary key field that uniquely identify a MongoDB
+document, override the ``$primaryKey`` property of the model class.
+
+By default, the model uses the PHP MongoDB driver to generate unique IDs for
+each document your Laravel application inserts.
+
+The following example specifies the ``name`` field as the primary key for
+the ``Planet`` class:
+
+.. literalinclude:: /includes/eloquent-models/PlanetPrimaryKey.php
+   :language: php
+   :emphasize-lines: 9
+   :dedent:
+
+To learn more about primary key behavior and customization options, see
+`Eloquent Primary Keys <https://laravel.com/docs/{+laravel-docs-version+}/eloquent#primary-keys>`__
+in the Laravel docs.
+
+To learn more about the ``_id`` field and the MongoDB document structure, see
+:manual:`Documents </core/document>` in the MongoDB server docs.
+
+.. _laravel-model-soft-delete:
+
+Enable Soft Deletes
+~~~~~~~~~~~~~~~~~~~
+
+Eloquent includes a soft delete feature that changes the behavior of the
+``delete()`` method on a model. When soft delete is enabled on a model, the
+``delete()`` method marks a document as deleted instead of removing it from the
+database.  It sets a timestamp on the ``deleted_at`` field to exclude it from
+retrieve operations automatically.
+
+To enable soft deletes on a class, import the ``MongoDB\Laravel\Eloquent\SoftDeletes``
+package and add the trait to the class definition as shown in the following
+code example:
```

Comment:
The `use` statement is just a syntax detail that is shown in the example.

```suggestion
To enable soft deletes on a class, add the trait ``MongoDB\Laravel\Eloquent\SoftDeletes``
to the class definition as shown in the following code example:
```

Reply:
Usually, we don't have `use` statement in our examples, so that thinks are more clear. And developers use cs tools to create the `use` statements for them.

Reply:
The `use` statement in the example is better.